### PR TITLE
Explainer: Multithreading

### PR DESCRIPTION
**Pull request by kainino0x**

### CODE REVIEW

File: explainer/index.bs [hunk 1]
```diff
@@ -713,7 +713,75 @@ When using advanced methods to transfer data to the GPU (with a rolling list of
     </pre>
 </div>
 
-## Multi-Threading ## {#multi-threading}
+## Multithreading ## {#multithreading}
+
+Multithreading is a key part of modern graphics APIs.
+Unlike OpenGL, newer APIs allow applications to encode commands, submit work, upload resources, and
```
Comment (Kangz) [inline explainer/index.bs:719]:
```suggestion
Unlike OpenGL, newer APIs allow applications to encode commands, submit work, transfer data to the GPU, and
```

**Review by Kangz — APPROVED** [review]

File: explainer/index.bs [hunk 2]
```diff
@@ -713,7 +713,75 @@ When using advanced methods to transfer data to the GPU (with a rolling list of
...
+This section describes the tentative plan for how it will work.
+
+As described in [[#gpu-process]], most WebGPU objects are actually just "handles" that refer to
+objects in the browser's GPU process.
+As such, it is relatively straightforward to allow these to be shared among threads.
+For example, a `GPUTexture` object can simply be `postMessage()`d to another thread, creating a
+new `GPUTexture` JavaScript object containing a handle to the *same* GPU process object.
+
+Several objects, like `GPUBuffer`, have client-side state.
+Applications still need to use them from multiple threads without `transfer`ring them back
```
Comment (kvark) [inline explainer/index.bs:734]: This is a bit unclear to me. Where are we controlling if something is transferred or not?

File: explainer/index.bs [hunk 3]
```diff
@@ -713,7 +713,75 @@ When using advanced methods to transfer data to the GPU (with a rolling list of
...
+
+Further discussion can be found in [#354](https://github.com/gpuweb/gpuweb/issues/354)
+(note not all of it reflects current thinking).
+
+### Unsolved: Synchronous Object Transfer ### {#multithreading-transfer}
+
+Some application architectures require objects to be passed between threads without having to
+asynchronously wait for a message to arrive on the receiving thread.
+
+The most crucial of these is WebAssembly applications:
```
Comment (kvark) [inline explainer/index.bs:767]: "is" -> "are"

File: explainer/index.bs (hunk 2 above)
Comment (kainino0x) [reply explainer/index.bs:734]: Transferability is WebIDL `[Transferable]` and is controlled via the `transfer` list in [`postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage).

**Review by kvark — APPROVED** [review]

**Pull request merged**